doc/src/content/xdocs/spec.xml

```diff
@@ -1052,17 +1061,21 @@
       <title>Schema Resolution</title>
 
       <p>A reader of Avro data, whether from an RPC or a file, can
-        always parse that data because its schema is provided.  But
-        that schema may not be exactly the schema that was expected.
+        always parse that data because the original schema must be
+        provided along with the data.  However, the reader may be
+        programmed to read data into a different schema.
         For example, if the data was written with a different version
-        of the software than it is read, then records may have had
-        fields added or removed.  This section specifies how such
+        of the software than it is read, then fields may have been
+        added or removed from records.  This section specifies how such
         schema differences should be resolved.</p>
 
-      <p>We call the schema used to write the data as
+      <p>We refer to the schema used to write the data as
         the <em>writer's</em> schema, and the schema that the
-        application expects the <em>reader's</em> schema.  Differences
-        between these should be resolved as follows:</p>
+        application expects the <em>reader's</em> schema.  In order
+        to resolve differences between the two, implementations must
+        generally first read serialized data with the writer's schema,
+        then convert it into the reader's schema, resolving differences
```

Comment:
Yes, that was the thought. I was trying to make the practical use of the reader & writer schemas & the need for resolution rules more apparent without being too specific about how a particular implementation really goes about it. If you would prefer a rewording I am happy to make adjustments.